From fe7e4a763a1e41d7ca5c2398f7cd950c8b67bb8f Mon Sep 17 00:00:00 2001
From: Richard Zhu <Richard.Zhu@freescale.com>
Date: Wed, 12 Nov 2014 09:42:20 +0800
Subject: [PATCH 1015/1074] MLK-9836-02 mcc: imx: do the error check after mcc
 send

add the error check after mcc_send is excuted.
end the tests when there is an error during mcc_send
or mcc_recv.

Signed-off-by: Richard Zhu <Richard.Zhu@freescale.com>
(cherry picked from commit c549c2a5ba57a300185f815c3a1b00c875e85fb7)
[Original patch taken from git://git.freescale.com/imx/linux-2.6-imx.git]
Signed-off-by: Biyao Zhai <biyao.zhai@windriver.com>
---
 drivers/char/imx_mcc/imx_mcc_test.c |    9 +++++++++
 1 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/drivers/char/imx_mcc/imx_mcc_test.c b/drivers/char/imx_mcc/imx_mcc_test.c
index 7f7a400..b0855cd 100644
--- a/drivers/char/imx_mcc/imx_mcc_test.c
+++ b/drivers/char/imx_mcc/imx_mcc_test.c
@@ -87,6 +87,10 @@ static ssize_t imx_mcc_can_test_en(struct device *dev,
 					&mcc_endpoint_a9_can, &msg,
 					sizeof(struct mcc_can_msg),
 					&num_of_received_bytes, 0xffffffff);
+			if (ret < 0) {
+				pr_err("A9 Main task recv error: %d\n", ret);
+				break;
+			}
 			pr_info("%s", msg.data);
 		}
 
@@ -147,6 +151,10 @@ static ssize_t imx_mcc_pingpong_en(struct device *dev,
 					&mcc_endpoint_m4_pingpong, &msg,
 					sizeof(struct mcc_pp_msg),
 					0xffffffff);
+			if (ret < 0) {
+				pr_err("A9 Main task send error: %d\n", ret);
+				break;
+			}
 
 			if (pingpong_en > 1) {
 				do_gettimeofday(&tv2);
@@ -184,6 +192,7 @@ static ssize_t imx_mcc_pingpong_en(struct device *dev,
 
 			if (MCC_SUCCESS != ret) {
 				pr_err("A9 Main task receive error: %d\n", ret);
+				break;
 			} else {
 				pr_info("%08x Main task received a msg"
 					" from [%d, %d, %d] endpoint\n", i,
-- 
1.7.5.4

